packages/server/src/utilities/workerRequests.ts

```diff
@@ -40,25 +40,21 @@ export function request(ctx?: Ctx, request?: any) {
 }
 
 async function checkResponse(
-  response: any,
+  response: Response,
   errorMsg: string,
   { ctx }: { ctx?: Ctx } = {}
 ) {
-  if (response.status !== 200) {
-    let error
-    try {
-      error = await response.json()
-      if (!error.message) {
-        error = JSON.stringify(error)
-      }
-    } catch (err) {
-      error = await response.text()
+  if (response.status >= 300) {
+    let responseErrorMessage
+    if (response.headers.get("content-type")?.includes("json")) {
+      const error = await response.json()
+      responseErrorMessage = error.message ?? JSON.stringify(error)
+    } else {
+      responseErrorMessage = await response.text()
     }
-    const msg = `Unable to ${errorMsg} - ${
-      error.message ? error.message : error
-    }`
+    const msg = `Unable to ${errorMsg} - ${responseErrorMessage}`
     if (ctx) {
-      ctx.throw(400, msg)
+      ctx.throw(msg, response.status)
```

Comment:
```suggestion
      ctx.throw(response.status, msg)
```

Reply:
It actually has to be the other way around to make use of the option status and default to 500:

```ts
/**
         * Throw an error with `msg` and optional `status`
         * defaulting to 500. Note that these are user-level
         * errors, and the message may be exposed to the client.
         *
         *    this.throw(403)
         *    this.throw('name required', 400)
         *    this.throw(400, 'name required')
         *    this.throw('something exploded')
         *    this.throw(new Error('invalid'), 400);
         *    this.throw(400, new Error('invalid'));
         *
         * See: https://github.com/jshttp/http-errors
         */
        throw(message: string, code?: number, properties?: {}): never;
```

Reply:
That... is absolutely terrifying.

Reply:
I've never seen anything like it to be fair 😂 

Funny code comment as well: https://github.com/jshttp/http-errors/blob/206aa2c15635dc1212c06c279540972aa90e23ea/index.js#L50